code/game/machinery/cryopod.dm

```diff
@@ -241,26 +243,42 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
 	container_resist(user)
 
 /obj/machinery/cryopod/process()
-	if(!occupant)
+	if(!occupant || ghost_offering)
 		return
 
 	var/mob/living/mob_occupant = occupant
 	if(mob_occupant)
-		// Eject dead people
-		if(mob_occupant.stat == DEAD)
+		// Eject people that are incapacitated
+		if(mob_occupant.stat)
 			open_machine()
 
+		//The grace period isn't up yet
 		if(!(world.time > despawn_world_time))
 			return
 
-		if(!mob_occupant.client && mob_occupant.stat < 2) //Occupant is living and has no client.
-			if(!control_computer_weakref)
-				find_control_computer(urgent = TRUE)//better hope you found it this time
+		//Offer special roles to ghosts and pause processing while we do
+		var/datum/antagonist/A = mob_occupant.mind.has_antag_datum(/datum/antagonist)
+		if(A)
+			ghost_offering = TRUE
+			INVOKE_ASYNC(src, PROC_REF(offering_to_ghosts), mob_occupant)
+			return
+		//This is not a role that needs to be offered, despawn them.
+		despawn_occupant()
 
-			despawn_occupant()
+/obj/machinery/cryopod/proc/offering_to_ghosts(mob/living/mob_occupant)
+	mob_occupant.SetUnconscious(30 SECONDS, TRUE)
+	if(offer_control(mob_occupant))
+		open_machine()
+	else
+		despawn_occupant()
```

Comment:
I think we need better recovery for when someone leaves but doesn't get taken by ghosts (perhaps it is early in the round or something).
My thinking would be something like this:
- Default antagonist behaviour would be to do nothing and let the cryo-pod keep pinging ghosts every 10 minutes until someone takes the role. (By returning false or something like that and letting the cryopod handle it form there)
- Overriding this behaviour would let you do stuff like transfer the role if possible or give it to someone else. (For example, blood brothers might convert the other brother into a traitor if nobody takes the role)

With the plans for antagonists to become important, we need to be able to error handle important characters leaving and not being immediately re-taken.